openedx/core/djangoapps/discussions/views.py

```diff
@@ -73,14 +97,36 @@ def to_representation(self, instance) -> dict:
                         for provider in instance.available_providers
                     },
                 },
-            }
+            })
             return payload
 
         def update(self, instance, validated_data):
             """
             Update and save an existing instance
             """
-            raise NotImplementedError
+            keys = [
```

Comment:
So is this the correct POST JSON payload?

```
{
  "enabled": <boolean>,
  "plugin_configuration": {}, // Think we don't need this today?
  "provider_type": "lti",
  "lti_configuration": {
    "lti_1p1_client_key": "abc123",
    "lti_1p1_client_secret": "321cba",
    "lti_1p1_launch_url": "https://localhost/launch"
  } 
}
```

Reply:
Yes. In fact, this is written to accept partial updates, so you really only need to pass in the parameters that are being changed (though you can still explicitly pass them all).